src/components/rooms-tab.js

```diff
@@ -127,11 +208,37 @@ function RoomsTab({ rooms }) {
       <section className="RoomsTab-section">
         <h2>{t('ownedRoomsHeader')}</h2>
         <Table dataSource={ownedRoomsRows} columns={ownedRoomsColumns} size="middle" />
+        <Restricted to={permissions.CREATE_ROOMS}>
+          <Button className="RoomsTab-newRoomButton" type="primary" shape="circle" icon={<PlusOutlined />} size="large" onClick={handleNewRoomClick} />
+        </Restricted>
       </section>
       <section className="RoomsTab-section">
         <h2>{t('joinedRoomsHeader')}</h2>
         <Table dataSource={membershipRoomsRows} columns={membershipRoomsColumns} size="middle" />
       </section>
+
+      {state.isNewRoomModalVisible && (
```

Comment:
I added this so that the modal "name" input field (value + validation message) gets reset when you:
- click the add room button
- invalidate the name field
- click 'cancel'
- click the add room button again

Due to the form validation, I found no other way for this. Maybe someone has a nicer solution?

Reply:
Yes, there is a nicer way by using the functions (like `resetFields` etc.) on the form instance (the ref you created), see also here: https://ant.design/components/form/#FormInstance

Reply:
Since only one field needed resetting (the other one functioned just fine with the `setState`) I checked the `Form.Item` documentation, but it makes sense for this to be on the `Form` level only 🤦‍♀️ 
Thanks, changed!
Kept it in the ok handler thought, feelt it's more future proof, for example if we'd change from redirecting after creation to just updating the rooms list.

Reply:
:+1:

Reply:
Had to make a small change to use `setFieldsValue` instead of `resetFields`, to reliably reset the `name` field.

Since the initial value is taken from the state, and the state change is async, sometimes the `name` was not changed when opening the modal.

Reply:
Alternatively you could not do anything when the modal opens, but reset the fields after each OK/Cancel click (on OK only after room creation, of course)